src/sort/sort.rs

```diff
@@ -185,26 +272,46 @@ fn exec(files: Vec<String>, settings: &Settings) -> i32 {
             }
         }
     }
-    else {
-        print_sorted(lines.iter(), &settings.outfile)
+    else if settings.merge {
+        if settings.unique {
+            print_sorted(file_merger.dedup(), &settings.outfile)
+        }
+        else {
+            print_sorted(file_merger, &settings.outfile)
+        }
+    }
+    else {        
+        if settings.unique {
```

Comment:
That doesn't work because `lines.iter().dedup()` is a `itertools::Dedup<std::slice::Iter<'_, std::string::String>>` but `lines.iter()` is `std::slice::Iter<'_, std::string::String>`